src/pal/src/include/pal/stackstring.hpp

```diff
@@ -114,6 +140,16 @@ class StackString
         return Set(s.m_buffer, s.m_count);
     }
 
+    //If takeOwnerShip set to TRUE- This object will take over the lifetime
+    //management of the buffer
+    void SetLiteral(const T * buffer, SIZE_T count, BOOL takeOwnerShip = FALSE)
```

Comment:
Actually, the only usage of the SetLiteral out of the Append, the one in the GetCurrentDirectoryA, passes in takeOwnership set to TRUE, so if you get rid of the use in the Append, no change other than adding SetLiteral without the takeOwnership should be needed.
And regarding the name, it seems confusing to me. "Literal" sounds like "constant" to me. "Acquire" might be a better name.